src/main/groovy/de/dkfz/eilslabs/batcheuphoria/execution/RestExecutionService.groovy

```diff
@@ -0,0 +1,207 @@
+        else
+            httpClient = HttpClientBuilder.create().build()
+
+        def httpRequest
+
+        if(restCommand.httpMethod == RestCommand.HttpMethod.HTTPPOST){
+            httpRequest = new HttpPost(url)
+            if(restCommand.requestBody)
+                httpRequest.setEntity(new StringEntity(restCommand.requestBody,"UTF-8"))
+        }else{
+            httpRequest = new HttpGet(url)
+        }
+
+        if(token){
+            restCommand.requestHeaders.add(new BasicHeader("Cookie", "platform_token=${token.replaceAll('"',"#quote#")}"))
+        }
+
+        if(restCommand.requestHeaders)
+            httpRequest.setHeaders((Header[])restCommand.requestHeaders.toArray())
+
+        CloseableHttpResponse response = httpClient.execute(httpRequest)
+
+        try {
+            logger.info("response status: "+response.getStatusLine())
+            HttpEntity entity = response.getEntity()
+            logger.info("response body: "+response.getEntity().contentType.value)
+            String result = IOUtils.toString(entity.content)
+            EntityUtils.consume(entity)
+            return new RestResult(statusCode: response.getStatusLine().getStatusCode(),body:result,headers:response.getAllHeaders())
+        } finally {
+            response.close()
+        }
+
+    }
+
+
+    private CloseableHttpClient createHttpClientWithClientCertAuth() {
+        try {
+            // read in the keystore from the filesystem, this should contain a single keypair
+            KeyStore clientKeyStore = KeyStore.getInstance("PKCS12");
+            clientKeyStore.load(new FileInputStream(this.keyStoreLocation), this.keyStorePassword.toCharArray());
+
+            SSLContext sslContext = SSLContexts
+                    .custom()
+                    .loadKeyMaterial(clientKeyStore, this.keyStorePassword.toCharArray())
+                    .loadTrustMaterial(null, new TrustSelfSignedStrategy())
+                    .build();
+
+            SSLConnectionSocketFactory sslConnectionFactory = new SSLConnectionSocketFactory(sslContext,
+                    new DefaultHostnameVerifier());
+
+            Registry<ConnectionSocketFactory> registry = RegistryBuilder.<ConnectionSocketFactory> create()
+                    .register("https", sslConnectionFactory)
+                    .register("http", new PlainConnectionSocketFactory())
+                    .build();
+
+            HttpClientBuilder builder = HttpClientBuilder.create();
+            builder.setSSLSocketFactory(sslConnectionFactory)
+            builder.setConnectionManager(new PoolingHttpClientConnectionManager(registry));
+
+            return builder.build();
+        } catch (Exception ex) {
+            // log ex
+            return null;
+        }
+    }
+
+    @NotYetImplemented
+    @Override
+    ExecutionResult execute(Command command) {
+        return null
+    }
+    @NotYetImplemented
+    @Override
+    ExecutionResult execute(Command command, boolean waitFor) {
```

Comment:
Reuse execute(Command)

Could you bundle the execute methods into one? With default parameters?